src/lib/rmt/wizard_firewall_page.rb

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) 2018 SUSE LLC.
+#  All Rights Reserved.
+
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of version 2 or 3 of the GNU General
+#  Public License as published by the Free Software Foundation.
+
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+#  GNU General Public License for more details.
+
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, contact SUSE LLC.
+
+#  To contact SUSE about this file by physical or electronic mail,
+#  you may find current contact information at www.suse.com
+
+require 'y2firewall/firewalld'
+require 'ui/event_dispatcher'
+require 'rmt/utils'
+require 'cwm/dialog'
+require 'cwm/custom_widget'
+
+Yast.import 'CWMFirewallInterfaces'
+
+module RMT; end
+
+class RMT::WizardFirewallPage < CWM::Dialog
+  def initialize
+    textdomain 'rmt'
+  end
+
+  def title
+    'RMT configuration: Firewall'
+  end
+
+  def contents
+    if firewalld.installed?
+      VBox(
+        VStretch(),
+        Label('For RMT to work properly, firewall ports for HTTP and HTTPS need to be opened.'),
+        VSpacing(),
+        RemoteFirewall.new,
+        VStretch()
+      )
+    else
+      VCenter(
+        Label("No firewall installed.\n\nIf you want to install and enable firewalld later,\nremember to open the firewall ports for HTTP and HTTPS.")
+      )
+    end
+  end
+
+  def run
+    if firewalld.installed?
+      Yast::Popup.ShowFeedback(_('Please wait'), _('Reading firewall settings ...')) do
+        firewalld.read
```

Comment:
you could move some logic to private methods to make the run method more readable

```ruby

def run
  read_config
  result = super
  write_config if result == :next
  result
end

private

def read_config
  return false if !firewalld.installed? || firewalld.read?
  Yast::Popup.Feedback(_('Please wait'), _('Reading firewall settings ...')) do
    firewalld.read
  end
end

def write_config
  return false if !firewalld.installed?
  ...
end

```